src/app/modules/products/components/product-detail/product-detail.component.ts

```diff
@@ -1,23 +1,60 @@
-import { Component, OnInit, Input } from "@angular/core";
+import { Component, Input } from "@angular/core";
+import { FormBuilder, Validators } from "@angular/forms";
+
 import { ImageService } from "@core/views/image.service";
 
 import { Product } from "@shared/models/classes/product.class";
+import { Weight } from "@shared/models/types/weight.type";
+import { Roast } from "@shared/models/types/roast.type";
+import { CustomOption } from "@shared/models/types/custom-option.interface";
 
 @Component({
     selector: "app-product-detail",
     templateUrl: "./product-detail.component.html",
     styleUrls: ["./product-detail.component.scss"]
 })
-export class ProductDetailComponent implements OnInit {
+export class ProductDetailComponent {
     @Input() product!: Product;
 
+    readonly weightOptions: Weight[] = ["250g", "1kg"];
+    readonly defaultWeight: Weight = this.weightOptions[0];
+    currentWeight: Weight = this.defaultWeight;
```

Comment:
this code is duplicated across components. The arrays of options should be moved to a constant file.

FYI, here is a cool trick to generate string union type from an array of strings:
`
const weightOptions = ["250g", "1kg"] as const;

export type Weight = typeof weightOptions[number];
`

so we should define the array of options in a const file (form-options.ts), then import the weight options and generate the type from the array. Solves duplicating the options in an array and a type, but also gives us type completion (should also be done in product-card